kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java

```diff
@@ -356,87 +361,34 @@ private void translateSymbols(Map<String, Boolean> attributes, SetMultimap<KLabe
-        if (koreAtt.contains("format")) {
-            formats.append("Lbl");
-            convert(prod.klabel().get().name(), formats);
-            formats.append('\n');
-            formats.append(koreAtt.get("format"));
-            formats.append('\n');
-        }
-        if (koreAtt.contains("color")) {
-            colors.append("Lbl");
-            convert(prod.klabel().get().name(), colors);
-            colors.append('\n');
-            for (int i = 0; i < prod.items().size(); i++) {
-                if (prod.items().apply(i) instanceof Terminal) {
-                    colors.append(koreAtt.get("color"));
-                    colors.append('\n');
-                }
-            }
-            if (koreAtt.contains("format")) {
-              String format = koreAtt.get("format");
-              boolean escape = false;
-              for (int i = 0; i < format.length(); i++) {
-                if (escape && format.charAt(i) == 'c') {
-                  colors.append(koreAtt.get("color"));
-                  colors.append('\n');
-                }
-                if (format.charAt(i) == '%') {
-                  escape = true;
-                } else {
-                  escape = false;
-                }
-              }
-            }
-            colors.append('\n');
-        } else if (koreAtt.contains("colors")) {
-            String[] data = koreAtt.get("colors").split(",");
-            colors.append("Lbl");
-            convert(prod.klabel().get().name(), colors);
-            colors.append('\n');
-            for (int i = 0; i < data.length; i++) {
-                colors.append(data[i].trim());
-                colors.append('\n');
-            }
-            colors.append('\n');
-        }
-        if (koreAtt.contains("assoc")) {
-            assocs.append("Lbl");
-            convert(prod.klabel().get().name(), assocs);
-            assocs.append('\n');
+    private void translateSymbol(Map<String, Boolean> attributes, SetMultimap<KLabel, Rule> functionRules, Set<KLabel> impurities, Set<Production> overloads, KLabel label, Production prod, StringBuilder sb) {
+        sb.append("  ");
+        if (isFunction(prod) && prod.att().contains(Att.HOOK()) && isRealHook(prod.att())) {
+            sb.append("hooked-");
         }
-        if (koreAtt.contains("comm")) {
-            comms.append("Lbl");
-            convert(prod.klabel().get().name(), comms);
-            comms.append('\n');
+        sb.append("symbol ");
+        convert(label, prod.params(), sb);
+        String conn;
+        sb.append("(");
+        conn = "";
+        for (NonTerminal nt : iterable(prod.nonterminals())) {
+            Sort sort = nt.sort();
+            sb.append(conn);
+            convert(sort, prod, sb);
+            conn = ", ";
         }
+        sb.append(") : ");
+        convert(prod.sort(), prod, sb);
+        sb.append(" ");
+        Att koreAtt = addKoreAttributes(prod, functionRules, impurities, overloads);
```

Comment:
So now instead of writing separate files with the unparsing information, we're adding attributes to the generated Kore file?

I like the approach more, just want to make sure I understand.

Reply:
I ended up needing extra information to support the bracket pass, so I decided to reevaluate the approach. The only reason I had used separate files originally was because it proved to be significantly faster. But I was able to optimize the code sufficiently that it became less of an issue, which is why I went back to my original idea.